bokeh/core/has_props.py

```diff
@@ -605,10 +605,16 @@ def apply_theme(self, property_values):
         elif hasattr(self, '__themed_values__'):
             delattr(self, '__themed_values__')
 
+        # Property container values might be cached even if unmodified. Invalidate
+        # any cached values that are not modified at this point.
+        for k, v in old_values.items():
+            if isinstance(v, PropertyValueContainer) and v._unmodified_default_value and k in self._property_values:
+                del self._property_values[k]
+
```

Comment:
Yes that seems right? Maybe could make a test case to catch this problem and then fix it... simplest could be to _unregister_owner right here I suppose.